### PR TITLE
Add logic for setting diet goals

### LINKED ISSUE #27: Add logic for setting goals
Goals class has been created. We need to create the logic for the users to set diet goals. Extension to issue #10

### PR DESCRIPTION
**Pull request by yicheng-toh**
Adding the ability for users to set diet goals based on our predefined nutrients.

Fixes #27 .

### CODE REVIEW

File: src/main/java/athleticli/ui/Parser.java [hunk 1]
```diff
@@ -353,6 +366,63 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
...
+            int targetValue;
+
+            ArrayList<DietGoal> dietGoals = new ArrayList<>();
+
+            for (int i = 0; i < nutrientAndTargetValues.length; i++) {
+                nutrientAndTargetValue = nutrientAndTargetValues[i].split("/");
+                nutrient = nutrientAndTargetValue[0];
+                targetValue = Integer.parseInt(nutrientAndTargetValue[1]);
+                System.out.println(nutrient);
+                System.out.println(targetValue);
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:393]: Are they debug output?

File: src/main/java/athleticli/ui/Parser.java [hunk 2]
```diff
@@ -353,6 +366,63 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
...
+    /**
+     * @param nutrient The nutrient that is provided by the user.
+     * @return boolean value depending on whether the nutrient is defined in our user guide.
+     *     It returns true if the nutrient is supported by our app, false otherwise.
+     */
+    public static boolean verifyValidNutrients(String nutrient) {
+        final String caloriesMarkerConstant = "calories";
+        final String proteinMarkerConstant = "protein";
+        final String carbMarkerConstant = "carb";
+        final String fatMarketConstant = "fat";
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:420]: Is it better to define them as static constants?

File: src/test/java/athleticli/ui/ParserTest.java [hunk 3]
```diff
@@ -169,4 +174,40 @@ void parseCommand_deleteDietCommand_emptyIndexExpectAthletiException() {
...
+        assertTrue(verifyValidNutrients("calories"));
+    }
+
+    @Test
+    void verifyNutrient_validNutrient_returnFalse() {
+        assertFalse(verifyValidNutrients("invalidNutrients"));
+    }
+
+    @Test
+    void parseDietGoalSet_oneValidGoal_oneGoalInList(){
```
Comment (skylee03) [inline src/test/java/athleticli/ui/ParserTest.java:189]:
```suggestion
    void parseDietGoalSet_oneValidGoal_oneGoalInList() {
```

File: src/test/java/athleticli/ui/ParserTest.java [hunk 4]
```diff
@@ -169,4 +174,40 @@ void parseCommand_deleteDietCommand_emptyIndexExpectAthletiException() {
...
+        String oneValidGoalString = "calories/60";
+        try {
+            assertEquals(1, parseDietGoalSet(oneValidGoalString).size());
+        } catch (AthletiException e){
+            assert(false);
```
Comment (skylee03) [inline src/test/java/athleticli/ui/ParserTest.java:194]: You can use `assertDoesNotThrow` instead!

File: src/main/java/athleticli/ui/Parser.java [hunk 5]
```diff
@@ -353,6 +366,63 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
...
+        return nutrient.equals(caloriesMarkerConstant) || nutrient.equals(proteinMarkerConstant)
+                || nutrient.equals(carbMarkerConstant) || nutrient.equals(fatMarketConstant);
+
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:423]:
```suggestion
```

File: src/main/java/athleticli/ui/Parser.java [hunk 6]
```diff
@@ -353,6 +366,63 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
...
+     * @param commandArgs User provided data to create goals for the nutrients defined.
+     * @return a list of diet goals for further checking in the Set Diet Goal Command.
+     * @throws AthletiException Invalid input by the user.
+     */
+    public static ArrayList<DietGoal> parseDietGoalSet(String commandArgs) throws AthletiException {
+        try {
+            String[] nutrientAndTargetValues;
+            if(commandArgs.contains(" ")) {
+                nutrientAndTargetValues = commandArgs.split("\\s+");
+            }else{
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:379]:
```suggestion
            } else {
```

File: src/main/java/athleticli/ui/Parser.java [hunk 7]
```diff
@@ -353,6 +366,63 @@ public static EditSleepCommand parseSleepEdit(String commandArgs) throws Athleti
...
 
+    /**
+     * @param commandArgs User provided data to create goals for the nutrients defined.
+     * @return a list of diet goals for further checking in the Set Diet Goal Command.
+     * @throws AthletiException Invalid input by the user.
+     */
+    public static ArrayList<DietGoal> parseDietGoalSet(String commandArgs) throws AthletiException {
+        try {
+            String[] nutrientAndTargetValues;
+            if(commandArgs.contains(" ")) {
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:377]:
```suggestion
            if (commandArgs.contains(" ")) {
```

File: src/main/java/athleticli/commands/diet/SetDietGoalCommand.java [hunk 8]
```diff
@@ -1,4 +1,55 @@
...
+            userNewNutrient = userNewDietGoals.get(i).getNutrients();
+            for (int j = 0; j < currentDietGoals.size(); j++) {
+                currentDietGoalsNutrient = currentDietGoals.get(j).getNutrients();
+                if (userNewNutrient.equals(currentDietGoalsNutrient)) {
+                    throw new AthletiException(String.format(
+                            Message.MESSAGE_DIETGOAL_ALREADY_EXISTED, userNewNutrient));
+                }
+            }
+        }
+        for (int k = 0; k < userNewDietGoals.size(); k++){
```
Comment (skylee03) [inline src/main/java/athleticli/commands/diet/SetDietGoalCommand.java:50]:
```suggestion
        for (int k = 0; k < userNewDietGoals.size(); k++) {
```

File: src/main/java/athleticli/ui/Parser.java [hunk 9]
```diff
@@ -3,34 +3,47 @@
...
+import java.util.ArrayList;
 
 /**
  * Defines the basic methods for command parser.
  */
 public class Parser {
+    private static final String caloriesMarkerConstant = "calories";
+    private static final String proteinMarkerConstant = "protein";
+    private static final String carbMarkerConstant = "carb";
+    private static final String fatMarketConstant = "fat";
```
Comment (skylee03) [inline src/main/java/athleticli/ui/Parser.java:39]: “Market” seems like a typo.
According to our [coding standard](https://se-education.org/guides/conventions/java/basic.html), constant names must be all uppercase using underscore to separate words.
```suggestion
    private static final String CALORIES_MARKER = "calories";
    private static final String PROTEIN_MARKER = "protein";
    private static final String CARB_MARKER = "carb";
    private static final String FAT_MARKER = "fat";
```

**Review by DaDevChia — COMMENTED** [review]
Good code quality, made a few suggestions to methods for better readability and logical flow

File: src/main/java/athleticli/commands/diet/SetDietGoalCommand.java [hunk 10]
```diff
@@ -1,4 +1,54 @@
...
+                    throw new AthletiException(String.format(
+                            Message.MESSAGE_DIETGOAL_ALREADY_EXISTED, userNewNutrient));
+                }
+            }
+        }
+        for (int k = 0; k < userNewDietGoals.size(); k++) {
+            currentDietGoals.add(userNewDietGoals.get(k));
+        }
+        return new String[]{"These are your goals:\n", currentDietGoals.toString()};
+    }
```
[the remark this answers is not in the document]
Comment (yicheng-toh) [reply src/main/java/athleticli/commands/diet/SetDietGoalCommand.java:53]: For this, would changing the underlying structure for diet goals be better?

File: src/main/java/athleticli/ui/Parser.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (yicheng-toh) [reply src/main/java/athleticli/ui/Parser.java:39]: Created issue #38

**Review by skylee03 — APPROVED** [review]
LGTM!

**Pull request merged**